File: Backend/omechoo/app/domain/entities/restaurant.py
from dataclasses import dataclass


@dataclass
class Location:
    """위치 정보"""
    latitude: float
    longitude: float
    address: str | None = None


@dataclass
class Restaurant:
    """식당 엔티티"""
    id: str
    name: str
    category: str
    location: Location | None = None
    phone: str | None = None
    rating: float | None = None
    
    # 웹사이트/지도 링크 등
    urls: list[str] | None = None
    
    # 메뉴 목록 (간단히 문자열 리스트로 관리)
    menu_items: list[str] | None = None

File: Backend/omechoo/frontend/src/types/restaurant.ts
import type { ApiResponse, Location } from './common';

export interface RestaurantSearchRequest {
  menu_id: string;
  latitude: number;
  longitude: number;
  radius_km?: number;
  max_result?: number;
}

export interface RestaurantCrawlRequest {
  url: string;
}

export interface LocationResponse extends Location {
  address?: string;
}

export interface Restaurant {
  id: string;
  name: string;
  category: string;
  location?: LocationResponse;
  urls?: string[];
  menu_items?: string[];
  distance?: number;
}

export interface RestaurantMenu {
  name: string;
  price: string;
}

export interface RestaurantDetail {
  rating: string;
  review_count: string;
  blog_review_count: string;
  business_status: any[];
  menus: RestaurantMenu[];
}

export type RestaurantSearchResponse = ApiResponse<Restaurant[]>;

export type RestaurantDetailResponse = ApiResponse<RestaurantDetail>;
